src/core/lint-hrefs-ids.js

```diff
@@ -0,0 +1,21 @@
+// Module core/lint-hrefs-ids
+// Warns about href's that do not have corresponding id's
+
+import { pub } from "core/pubsubhub";
+
+export const name = "core/lint-hrefs-ids";
+
+export function run(conf, doc, cb) {
+  console.log()
+  Array.from(doc.querySelectorAll('a[href]'))
+    .map(a => ({ a, href: a.getAttribute('href') }))
+    .filter(({ href }) => Boolean(href))
+    .filter(({ href }) => href.startsWith('#'))
+    .map(({ a, href }) => ({ a, href, id: href.slice(1) }))
+    .filter(({ id }) => Boolean(id))
+    .filter(({ id }) => !doc.getElementById(id))
+    .forEach(({ href, id }) => {
+      pub("warn", `Found link with href '${href}' but no element with id '${id}'`);
```

Comment:
here we actually do want to `console.warn(a)` in addition to `pub()`. 

Then the message is:  `Found `a` element with href '${href}' but no element with id '${id}'. See developer console for offending element.`

You can also do: `a.classList.add("respec-offending-element")`, which will add a red underline to the offending element.